tests/trainer/test_trainer.py

```diff
@@ -98,6 +99,52 @@ def test_optimizer_params_on_device(
         target_device = 'cuda' if device == 'gpu' else 'cpu'
         assert all(param.device.type == target_device for param in parameters)
 
+    @pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.0.0'),
```

Comment:
can we also add an FSDP + compile test please?

Reply:
Done in https://github.com/mosaicml/composer/pull/2118/commits/d0e8481b1a3de2bb20239802a9adfac217fb45cc. 

**Test output:**

```
collected 6 items

tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[None-Precision.FP32] PASSED                                                                          [ 16%]
tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[None-Precision.AMP_BF16] PASSED                                                                      [ 33%]
tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[None-Precision.AMP_FP16] PASSED                                                                      [ 50%]
tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[compile_config1-Precision.FP32] PASSED                                                               [ 66%]
tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[compile_config1-Precision.AMP_BF16] PASSED                                                           [ 83%]
tests/trainer/test_trainer.py::TestTrainerInitOrFit::test_fsdp[compile_config1-Precision.AMP_FP16] PASSED                                                           [100%]

============================================================================ 6 passed in 5.55s ============================================================================
```